sdk/src/Core/Internal/Utils/IPEndPointExtension.cs

```diff
@@ -143,21 +147,27 @@ public static bool TryParse(string input, out HostEndPoint hostEndpoint)
         /// <returns>true if <paramref name="input"/> converted successfully; otherwise, false.</returns>
         public static bool TryParse(string input, out EndPoint endpoint)
         {
-            if (TryParse(input, out IPEndPoint ipEndPoint))
+            endpoint = null;
+            if (IsIPAddress(input))
             {
-                endpoint = EndPoint.Of(ipEndPoint);
-                return true;
-            }
-            else if (TryParse(input, out HostEndPoint hostEndPoint))
-            {
-                endpoint = EndPoint.Of(hostEndPoint);
-                return true;
+                _logger.DebugFormat("Determined that {0} is an IP.", input);
+                if (TryParse(input, out IPEndPoint ipEndPoint))
+                {
+                    endpoint = EndPoint.Of(ipEndPoint);
+                    return true;
+                }
             }
             else
             {
-                endpoint = null;
-                return false;
+                _logger.DebugFormat("Determined that {0} is not an IP.", input);

```

Comment:
can we modify this message to ` _logger.DebugFormat("Determined that {0} is not an IP address, would try to parse it as hostname.", input);`

Absence of this may confuse the user.

Reply:
Addressed in 0de9693.